src/docs/platform-automation/pruning.md

```diff
@@ -0,0 +1,103 @@
+
+audience: technical lead, developer
+
+author: Steven Barre
+
+content_owner: Olena Mitovska
+
+sort_order: 4
+---
+
+# Pruning
+Last updated: **March 6, 2024**
+
+There are a few CronJobs set up on the OpenShift clusters to help prune old objects and keep the cluster clean. This page discusses each of the different object types and how and when they are pruned.
+
+## On this page
+* **[DeploymentConfigs](#deploymentconfigs)**
+* **[Builds](#builds)**
+* **[ImageStreams](#imagestreams)**
+* **[Pipelines](#pipelines)**
+* **[Related pages](#related-pages)**
+
+## DeploymentConfigs
+
+When a `DeploymentConfig` rolls out a new version, it creates a new `ReplicationController`. After some time, the number of these can pile up. So older ones are pruned daily.
+
+A few settings are passed to the pruner job:
+
+* Per `DeploymentConfig` object, keep the last 2 `ReplicationControllers` that have a status of `Complete` and replica count of zero.
+* Per `DeploymentConfig` object, keep the last 1 `ReplicationControllers` that have a status of `Failed` and replica count of zero.
+* Per `DeploymentConfig` object, keep any `ReplicationControllers` that is younger than 48 hours.
+* Prune all `ReplicationControllers` that no longer have a `DeploymentConfig` object, has status of `Complete` or `Failed`, and has a replica count of zero.
+
+## Builds
+
+When a `BuildConfig` runs a build a new `Build` object is created along with the pods that the build runs in. After some time, the number of these can pile up. So older ones are pruned daily.
+
+A few settings are passed to the pruner job:
+
+* Prune all builds whose build configuration no longer exists, status is complete, failed, error, or canceled.
+* Per `BuildConfig` object, keep the last 2 `Builds` whose status is `Complete`.
+* Per `BuildConfig` object, keep the last 1 `Builds` whose status is `Failed`, `Error`, or `Canceled`.
+* Per `BuildConfig` object, keep any `Builds` that is younger than 48 hours.
+
+## ImageStreams
+
+### Daily Prune
+
+As new images are built and pushed to tags, the older copies of that tag are retained. Each of these is called a revision and is tracked with a `sha256`. After some time, the number of these can pile up. So older ones are pruned daily.
+
+A few settings are passed to the pruner job:
+
+* All revisions created in the last 48h are kept
+* All revisions in use by a running pod, or in a deployment are kept. See [the docs](https://docs.openshift.com/container-platform/4.13/applications/pruning-objects.html#pruning-images-conditions_pruning-objects) for a full list
+* Only the most recent revision is kept
+* All other revisions are pruned
+
+### Image Registry Size warnings
+
+The platform will now be sending a weekly email to teams that are using too much space on the registry. The image registry is a shared service and overuse of it can lead to other teams being unable to push their builds, or to the platform team having to buy more storage space.
```

Comment:
```suggestion
The platform sends out a weekly email to teams utilizing an too much space on the shared image registry. This registry is a shared service, and if it's overused, it can result in other teams being unable to push their builds. Alternatively, it may need the platform team to purchase additional storage space.
```